fish

```diff
@@ -0,0 +1,33 @@
+# All pitfalls are taken from https://fishshell.com/docs/current/faq.html
+
+# Creating a variable:
+set -x variable value # typically set -gx key value
+
+# Removing a variable:
+set -e variable
+
+# Checking whether variable is defined:
+if set -q variable
+    some_command
+end
+
+# Checking whether variable is not empty:
+if string length -q -- $variable
+   some_command
+end
+
+# Choosing `coolbeans` theme in an interactive shell:
+if status is-interactive
+    fish_config theme choose coolbeans
+end
```

Comment:
This makes no sense as presented here without context.  The default assumption is that you are in the tty and it is an interactive session. This would normally also be a one-time thing (you would run `fish_config theme save` afterwards to persist the setting) and probably doesn't belong in a cheat-sheet.

Reply:
I've removed this example.